### PR TITLE
3565 - adds metadata when loading dicom series

### LINKED ISSUE #3565: metadata when loading DICOM series
**Is your feature request related to a problem? Please describe.**
the image reader ignores metadata when loading from a directory using itk:
```py
import itk
from monai.transforms import LoadImage

loader = LoadImage(reader="ITKReader", pixel_type=itk.UC)
img, metadata = loader("tests/testing_data/CT_DICOM")
print(metadata.keys())
```

output:
```
dict_keys(['spacing', 'original_affine', 'affine', 'spatial_shape', 'original_channel_dim', 'filename_or_obj'])
```

**Describe the solution you'd like**
the metadata output should include the dicom tags available:
```
dict_keys(['0008|0005', '0008|0008', '0008|0012', '0008|0013', '0008|0014', '0008|0016', '0008|0018', '0008|0020', '0008|0021', '0008|0022', '0008|0023', '0008|0030', '0008|0031', '0008|0032', '0008|0033', '0008|0050', '0008|0060', '0008|0061', '0008|0070', '0008|0090', '0008|0201', '0008|1030', '0008|103e', '0008|1090', '0010|0010', '0010|0020', '0010|0030', '0010|0040', '0010|1010', '0012|0062', '0012|0063', '0018|0015', '0018|0022', '0018|0050', '0018|0060', '0018|0090', '0018|1020', '0018|1030', '0018|1100', '0018|1110', '0018|1111', '0018|1120', '0018|1130', '0018|1140', '0018|1150', '0018|1151', '0018|1152', '0018|1170', '0018|1190', '0018|1210', '0018|5100', '0020|000d', '0020|000e', '0020|0010', '0020|0011', '0020|0012', '0020|0013', '0020|0032', '0020|0037', '0020|0052', '0020|1040', '0020|1041', '0028|0002', '0028|0004', '0028|0010', '0028|0011', '0028|0030', '0028|0100', '0028|0101', '0028|0102', '0028|0103', '0028|0120', '0028|1050', '0028|1051', '0028|1052', '0028|1053', '0032|000a', '0032|000c', '0032|1030', '0032|1033', '0040|0244', '0040|0245', '0040|0254', '0040|1001', 'spacing', 'original_affine', 'affine', 'spatial_shape', 'original_channel_dim', 'filename_or_obj'])
```

### PR DESCRIPTION
**Pull request by wyli**
Signed-off-by: Wenqi Li <wenqil@nvidia.com>

Fixes #3565

### Description
adds an option `series_meta: bool`

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [x] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [x] In-line docstrings updated.
- [x] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Comment by Nic-Ma** [comment]
Hi @wyli ,

Thanks for your enhancement, is this PR related to: https://github.com/Project-MONAI/MONAI/discussions/3564?

**Review by Nic-Ma — APPROVED** [review]
Looks good to me, put some minor comments inline.

Thanks.

File: monai/data/image_reader.py [hunk 1]
```diff
@@ -221,7 +229,17 @@ def read(self, data: Union[Sequence[PathLike], PathLike], **kwargs):
                 series_identifier = series_uid[0] if not self.series_name else self.series_name
                 name = names_generator.GetFileNames(series_identifier)
 
-            img_.append(itk.imread(name, **kwargs_))
+                _obj = itk.imread(name, **kwargs_)
+                if self.series_meta:
+                    _reader = itk.ImageSeriesReader.New(FileNames=name)
```
[the remark this answers is not in the document]
Comment (wyli) [reply monai/data/image_reader.py:234]: This flag is checked only when loading DICOM series, I think the docstring describes this behaviour

File: monai/data/image_reader.py [hunk 2]
```diff
@@ -159,19 +159,27 @@ class ITKReader(ImageReader):
             If ``False``, the spatial indexing follows the numpy convention;
             otherwise, the spatial indexing convention is reversed to be compatible with ITK. Default is ``False``.
             This option does not affect the metadata.
+        series_meta: whether to load the metadata of the DICOM series (using the metadata from the first slice).
+            This flag is checked only when loading DICOM series.
```
[the remark this answers is not in the document]
Comment (wyli) [reply monai/data/image_reader.py:163]: thanks, this is updated.

File: monai/data/image_reader.py [hunk 3]
```diff
@@ -221,7 +229,17 @@ def read(self, data: Union[Sequence[PathLike], PathLike], **kwargs):
...
                 name = names_generator.GetFileNames(series_identifier)
 
-            img_.append(itk.imread(name, **kwargs_))
+                _obj = itk.imread(name, **kwargs_)
+                if self.series_meta:
+                    _reader = itk.ImageSeriesReader.New(FileNames=name)
+                    _reader.Update()
+                    _meta = _reader.GetMetaDataDictionaryArray()
+                    if len(_meta) > 0:
+                        # using the first slice's meta. this could be improved to filter unnecessary tags.
```
[the remark this answers is not in the document]
Comment (wyli) [reply monai/data/image_reader.py:238]: thanks, this is updated.

**Pull request merged**